bmp-deploy/doc/config.md
# [BMP Installation Guide](main.md) - Configuration Management
The `.env` file contains various configuration parameters for BMP.
~~~
BMP_HOST_IP=xxx.xxx.xxx.xxx
BMP_VERSION=dev
COMPOSE_PROJECT_NAME=bmp
BMP_DB_HOST=bmp-db
BMP_DB_PORT=3306
BMP_DB_USER=bmp_rw
BMP_DB_PASSWORD='LpK9Jq12Zf'
BMP_DB_NAME=bmp
BMP_REDIS_HOST=bmp-redis
BMP_REDIS_PORT=6379
BMP_REDIS_PASSWORD='LpK9Jq12Zf'
BMP_MQ_HOST=bmp-mq
BMP_MQ_PORT=5672
BMP_MQ_USER=bmp
BMP_MQ_PASSWORD='LpK9Jq12Zf'
BMP_MQ_VHOST=/bmp
BMP_OMAPI_HOST=bmp-dhcp-agent
BMP_OMAPI_PORT=7911
BMP_OMAPI_TOKEN="LpK9Jq12Zf"
BMP_OPENAPI_HOST=bmp-openapi
BMP_OPENAPI_PORT=8801
BMP_OPENAPI_CONSOLE_HOST=bmp-openapi-console
BMP_OPENAPI_CONSOLE_PORT=8802
BMP_OOB_ALERT_HOST=bmp-oob-alert
BMP_OOB_ALERT_PORT=8804
BMP_MQ_EXCHANGE_ROUTING_KEY=idc-vm4xsulx1k2d9z4xkctrttig02zl
BMP_IMAGE_PORT=10000
BMP_MONITOR_PROXY_PORT=8805
BMP_PROMETHEUS_HOST=bmp-prometheus
BMP_PROMETHEUS_PORT=9090
BMP_PROMETHEUS_DATA_DIR=/data/prometheus
BMP_PUSHGATEWAY_HOST=bmp-pushgateway
BMP_PUSHGATEWAY_PORT=9091
BMP_ALERTMANAGER_HOST=bmp-alertmanager
BMP_ALERTMANAGER_PORT=9093
BMP_PRONOEA_HOST=bmp-pronoea
BMP_PRONOEA_PORT=9999
~~~

## Configuration Explanation:
| Configuration Name                         | Description                                            |
|-------------------------------------------|--------------------------------------------------------|
| BMP_HOST_IP                 | Manager node IP, please configure it as the management network card IP, BMP various service components listen on this IP         |
| BMP_VERSION                 | The version of BMP to be installed, please do not modify it randomly                              |
| COMPOSE_PROJECT_NAME        | Docker Compose project name, default value is bmp, please do not modify               |
| BMP_DB_HOST                 | BMP-DB IP/hostname, please do not modify                             |
| BMP_DB_PORT                 | BMP-DB port, please do not modify                                |
| BMP_DB_USER                 | BMP-DB username                                    |
| BMP_DB_PASSWORD             | BMP-DB user password                                   |
| BMP_DB_NAME                 | BMP-DB database name, please do not modify                              |
| BMP_REDIS_HOST              | BMP-Redis IP/hostname, please do not modify                          |
| BMP_REDIS_PORT              | BMP-Redis port, please do not modify                             |
| BMP_REDIS_PASSWORD          | BMP-Redis password                                  |
| BMP_MQ_HOST                 | BMP-MQ IP/hostname, please do not modify                             |
| BMP_MQ_PORT                 | BMP-MQ port, please do not modify                                |
| BMP_MQ_USER                 | BMP-MQ username                                    |
| BMP_MQ_PASSWORD             | BMP-MQ password                                     |
| BMP_MQ_VHOST                | BMP-MQ vhost name, please do not modify                            |
| BMP_OMAPI_HOST              | DHCP OMAPI service IP/hostname, please do not modify                       |
| BMP_OMAPI_PORT              | DHCP OMAPI service port, please do not modify                           |
| BMP_OMAPI_TOKEN             | DHCP OMAPI authentication token                             |
| BMP_OPENAPI_HOST            | BMP-OpenAPI IP/hostname, please do not modify                        |
| BMP_OPENAPI_PORT            | BMP-OpenAPI port, please do not modify                           |
| BMP_OPENAPI_CONSOLE_HOST    | BMP-OpenAPI-Console IP/hostname, please do not modify                |
| BMP_OPENAPI_CONSOLE_PORT    | BMP-OpenAPI-Console port, please do not modify                   |
| BMP_OOB_ALERT_HOST          | BMP-OOB-Alert IP/hostname, please do not modify                      |
| BMP_OOB_ALERT_PORT          | BMP-OOB-Alert port, please do not modify                         |
| BMP_MQ_EXCHANGE_ROUTING_KEY | MQ routing key, this value should be equal to idc_id, please do not modify             |
| BMP_IMAGE_PORT              | BMP image server port, please do not modify                             |
| BMP_MONITOR_PROXY_PORT      | BMP-monitor-proxy port, please do not modify                     |
| BMP_PROMETHEUS_HOST         | BMP-Prometheus IP/hostname, please do not modify                     |
| BMP_PROMETHEUS_PORT         | BMP-Prometheus port, please do not modify                        |
| BMP_PROMETHEUS_DATA_DIR     | BMP-Prometheus data storage path, this path requires a large amount of storage space, please modify according to actual needs |
| BMP_PUSHGATEWAY_HOST        | BMP-Pushgateway IP/hostname, please do not modify                    |
| BMP_PUSHGATEWAY_PORT        | BMP-Pushgateway port, please do not modify                       |
| BMP_ALERTMANAGER_HOST       | BMP-Alertmanager IP/hostname, please do not modify                   |
| BMP_ALERTMANAGER_PORT       | BMP-Alertmanager port, please do not modify                      |
| BMP_PRONOEA_HOST            | BMP-Pronoea IP/hostname, please do not modify                        |
| BMP_PRONOEA_PORT            | BMP-Pronoea port, please do not modify                           |

## Note
* The values of BMP_DB_PASSWORD, BMP_REDIS_PASSWORD, BMP_MQ_PASSWORD, and BMP_OMAPI_TOKEN can only consist of **uppercase and lowercase letters and digits**. Including special characters may cause errors.

docs/bmp-deploy/config.md
#  [bmp安装指导](main.md) - 配置管理
.env file configuration
~~~
BMP_HOST_IP=xxx.xxx.xxx.xxx
# bmp registry
BMP_REGISTRY=docker.io

# db middleware registry eg: mysql:5.7-debian
# if you cannot pull from docker.io please change docker.io to other Container image repository
BASE_REGISTRY=docker.m.daocloud.io

BMP_VERSION=dev
COMPOSE_PROJECT_NAME=bmp
BMP_DB_HOST=bmp-db
BMP_DB_PORT=3306
BMP_DB_USER=bmp_rw
BMP_DB_PASSWORD='LpK9Jq12Zf'
BMP_DB_NAME=bmp
BMP_REDIS_HOST=bmp-redis
BMP_REDIS_PORT=6379
BMP_REDIS_PASSWORD='LpK9Jq12Zf'
BMP_MQ_HOST=bmp-mq
BMP_MQ_PORT=5672
BMP_MQ_USER=bmp
BMP_MQ_PASSWORD='LpK9Jq12Zf'
BMP_MQ_VHOST=/bmp
BMP_OMAPI_HOST=bmp-dhcp-agent
BMP_OMAPI_PORT=7911
BMP_OMAPI_TOKEN="LpK9Jq12Zf"
BMP_OPENAPI_HOST=bmp-openapi
BMP_OPENAPI_PORT=8801
BMP_OPENAPI_CONSOLE_HOST=bmp-openapi-console
BMP_OPENAPI_CONSOLE_PORT=8802
BMP_OOB_ALERT_HOST=bmp-oob-alert
BMP_OOB_ALERT_PORT=8804
BMP_MQ_EXCHANGE_ROUTING_KEY=idc-vm4xsulx1k2d9z4xkctrttig02zl
BMP_IMAGE_PORT=10000
BMP_MONITOR_PROXY_PORT=8805
BMP_PROMETHEUS_HOST=bmp-prometheus
BMP_PROMETHEUS_PORT=9090
BMP_PROMETHEUS_DATA_DIR=/data/prometheus
BMP_PUSHGATEWAY_HOST=bmp-pushgateway
BMP_PUSHGATEWAY_PORT=9091
BMP_ALERTMANAGER_HOST=bmp-alertmanager
BMP_ALERTMANAGER_PORT=9093
BMP_PRONOEA_HOST=bmp-pronoea
BMP_PRONOEA_PORT=9999
~~~

## Configuration Description:
| Configuration Name | Description |
|---|---|
| BMP_HOST_IP | IP of the manager node, please configure it to the management network card IP, BMP service components listen on this IP. |
| BMP_VERSION | Version of BMP to be installed, do not modify casually. |
| COMPOSE_PROJECT_NAME | Docker Compose project name, default value is `bmp`, do not modify. |
| BMP_DB_HOST | IP/hostname of bmp-db, do not modify. |
| BMP_DB_PORT | Port of bmp-db, do not modify. |
| BMP_DB_USER | Username for bmp-db. |
| BMP_DB_PASSWORD | Password for bmp-db. |
| BMP_DB_NAME | Database name for bmp-db, do not modify. |
| BMP_REDIS_HOST | IP/hostname of bmp-redis, do not modify. |
| BMP_REDIS_PORT | Port of bmp-redis, do not modify. |
| BMP_REDIS_PASSWORD | Password for bmp-redis. |
| BMP_MQ_HOST | IP/hostname of bmp-mq, do not modify. |
| BMP_MQ_PORT | Port of bmp-mq, do not modify. |
| BMP_MQ_USER | Username for bmp-mq. |
| BMP_MQ_PASSWORD | Password for bmp-mq. |
| BMP_MQ_VHOST | Vhost name for bmp-mq, do not modify. |
| BMP_OMAPI_HOST | IP/hostname of DHCP OMAPI service, do not modify. |
| BMP_OMAPI_PORT | Port of DHCP OMAPI service, do not modify. |
| BMP_OMAPI_TOKEN | Authentication token for DHCP OMAPI. |
| BMP_OPENAPI_HOST | IP/hostname of bmp-openapi, do not modify. |
| BMP_OPENAPI_PORT | Port of bmp-openapi, do not modify. |
| BMP_OPENAPI_CONSOLE_HOST | IP/hostname of bmp-openapi-console, do not modify. |
| BMP_OPENAPI_CONSOLE_PORT | Port of bmp-openapi-console, do not modify. |
| BMP_OOB_ALERT_HOST | IP/hostname of bmp-oob-alert, do not modify. |
| BMP_OOB_ALERT_PORT | Port of bmp-oob-alert, do not modify. |
| BMP_MQ_EXCHANGE_ROUTING_KEY | MQ routing key, this value should be equal to `idc_id`, do not modify. |
| BMP_IMAGE_PORT | Port of BMP image server, do not modify. |
| BMP_MONITOR_PROXY_PORT | Port of bmp-monitor-proxy, do not modify. |
| BMP_PROMETHEUS_HOST | IP/hostname of bmp_prometheus, do not modify. |
| BMP_PROMETHEUS_PORT | Port of bmp_prometheus, do not modify. |
| BMP_PROMETHEUS_DATA_DIR | Data storage path for bmp_prometheus, this path requires a large amount of storage space, it is recommended to modify according to actual conditions. |
| BMP_PUSHGATEWAY_HOST | IP/hostname of bmp_pushgateway, do not modify. |
| BMP_PUSHGATEWAY_PORT | Port of bmp_pushgateway, do not modify. |
| BMP_ALERTMANAGER_HOST | IP/hostname of bmp_alertmanager, do not modify. |
| BMP_ALERTMANAGER_PORT | Port of bmp_alertmanager, do not modify. |
| BMP_PRONOEA_HOST | IP/hostname of bmp_pronoea, do not modify. |
| BMP_PRONOEA_PORT | Port of bmp_pronoea, do not modify. |

## Notes
* The values of `BMP_DB_PASSWORD`, `BMP_REDIS_PASSWORD`, `BMP_MQ_PASSWORD`, and `BMP_OMAPI_TOKEN` can only consist of **uppercase and lowercase letters and numbers**. Special characters may cause errors.






bmp-agent/Changelog.md
# Changelog

## [2.0.7] - 2024-08-09
### Added
- 新增日志同步到rsyslog
- 修复format分区顺序和partition不一致时导致分区显示乱序问题
- 新增Ubuntu22.04版本ramos
- 新增Ubuntu20.04版本ramos
- 修复cannot import name 'MutableMapping' from 'collections'

## [2.0.6] - 2024-08-05
### Fixed
- 修改刷新数据到硬盘的方式

## [2.0.5] - 2024-08-05
### Fixed
- 修复写入文件的数据未刷到硬盘

## [2.0.4] - 2024-08-02
### Changed
- 采集信息返回网卡bus_info

## [2.0.3] - 2024-07-22
### Added
- 修复硬盘发现较慢导致makeraid的sn值获取错误

## [2.0.2] - 2024-07-19
### Added
- 采集信息指令新增返回系统架构字段

## [2.0.1] - 2024-05-28
### Added
- RamOS构建新增版本号
- 新增Ubuntu22.04RamOS

### Changed
- 修改采集指令
- 优化raid指令

## [2.0.0] - 2023-03-03
### Added
- 升级到Python3，接口有重大调整，不兼容老系统

bmp-dhcp-agent/bmpda/config.py
import os

from bmpda import constants
from bmpda.utils import utils

env: str = os.getenv('ENV')
if env:
    cfg_path = os.path.join('cfg', env)
    AGENT_PARAMS: dict = utils.get_agent_params(
        utils.get_relative_path(cfg_path))
else:
    AGENT_PARAMS = os.environ
heart_period: int = int(AGENT_PARAMS.get('BMP_HEART_PERIOD', 10))
mq_host: str = AGENT_PARAMS.get('BMP_MQ_HOST')  # required
mq_port: int = int(AGENT_PARAMS.get('BMP_MQ_PORT', 5672))
mq_username: str = AGENT_PARAMS.get('BMP_MQ_USER')  # required
mq_password: str = AGENT_PARAMS.get('BMP_MQ_PASSWORD')  # required
mq_password = mq_password.strip("'")
mq_vhost: str = AGENT_PARAMS.get('BMP_MQ_VHOST', '/bmp')
mq_exchange: str = AGENT_PARAMS.get('BMP_MQ_EXCHANGE', 'BMP_SCHEDULER')
mq_exchange_type: str = AGENT_PARAMS.get('BMP_MQ_EXCHANGE_TYPE', 'topic')
mq_exchange_type_publish: str = AGENT_PARAMS.get(
    'BMP_MQ_EXCHANGE_TYPE_PUBLISH', 'direct')
mq_exchange_routing_key: str = AGENT_PARAMS.get(
    'BMP_MQ_EXCHANGE_ROUTING_KEY', 'az')
mq_exchange_consumer: str = AGENT_PARAMS.get(
    'BMP_MQ_EXCHANGE_CONSUMER', 'BMP_SCHEDULER_TOPIC')
mq_queue_name_pefix_consumer: str = AGENT_PARAMS.get(
    'BMP_MQ_QUEUE_NAME_PEFIX_CONSUMER', 'dhcp-az')
omapi_host = AGENT_PARAMS.get('BMP_OMAPI_HOST', '127.0.0.1')
omapi_port = AGENT_PARAMS.get('BMP_OMAPI_PORT', 7911)
omapi_key = AGENT_PARAMS.get('BMP_OMAPI_KEY', '')
dhcp_config_dir = AGENT_PARAMS.get('BMP_DHCP_CONFIG_DIR', '/data')
dhcp_control_bin_path = utils.get_relative_path('shell/dhcpd_control.sh')
dhcp_control_bin = AGENT_PARAMS.get(
    'BMP_DHCP_CONTROL_BIN', dhcp_control_bin_path)
protocol: str = AGENT_PARAMS.get('BMP_PROTOCAL', constants.PROTOCAL_RABBITMQ)
log_path: str = AGENT_PARAMS.get('BMP_LOG_PATH',
                                 '/var/log/bmp/bmp-dhcp-agent/bmp-dhcp-agent.log')
# DEBUG,INFO,WARNING
logging_level = AGENT_PARAMS.get('LOGGING_LEVEL', 'INFO')

# tftp
tftp_config_dir = AGENT_PARAMS.get('BMP_TFTP_CONFIG_DIR', '/data')
tftp_mq_host: str = AGENT_PARAMS.get('BMP_TFTP_MQ_HOST')
tftp_mq_port: int = int(AGENT_PARAMS.get('BMP_TFTP_MQ_PORT', 5672))
tftp_mq_user: str = AGENT_PARAMS.get('BMP_TFTP_MQ_USER')
tftp_mq_password: str = AGENT_PARAMS.get('BMP_TFTP_MQ_PASSWORD')
tftp_mq_password = tftp_mq_password.strip("'")
tftp_mq_vhost: str = AGENT_PARAMS.get('BMP_TFTP_MQ_VHOST', '/bmp')
tftp_mq_exchange_routing_key: str = AGENT_PARAMS.get(
    'BMP_TFTP_MQ_EXCHANGE_ROUTING_KEY', '')
tftp_image_host: str = AGENT_PARAMS.get('BMP_TFTP_IMAGE_HOST', '')
tftp_image_port: str = AGENT_PARAMS.get('BMP_TFTP_IMAGE_PORT', '')
tftp_rsyslog_host: str = AGENT_PARAMS.get('BMP_TFTP_RSYSLOG_HSOT', '')
tftp_rsyslog_port: str = AGENT_PARAMS.get('BMP_TFTP_RSYSLOG_PORT', '')
